src/SoftApService.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+
+import type { WifiNetwork } from './types';
+
+import { fetchJSON } from './stores/ApiRequestStores/makeRequestApiStore';
+import NodeRSA from 'node-rsa';
+
+const BASE_URL = 'http://192.168.0.1:80';
+const SUCCESS_RESPONSE_CODE = 0;
+const DEFAULT_WIFI_INDEX = 0;
+const DEFAULT_WIFI_CHANNEL = 3;
+
+/* eslint-disable sorting/sort-object-props */
+export const WIFI_SECURITIES = {
+  OPEN: 0,
+  WEP_PSK: 1,
+  WEP_SHARED: 32769,
+  WPA_AES_PSK: 2097156,
+  WPA_TKIP_PSK: 2097154,
+  WPA2_AES_PSK: 4194308,
+  WPA2_MIXED_PSK: 4194310,
+  WPA2_TKIP_PSK: 4194306,
+};
+/* eslint-enable */
+
+const translateWifiFromApi = ({
+  ch,
+  mdr,
+  rssi,
+  sec,
+  ssid,
+}: Object): WifiNetwork => ({
+  channel: ch,
+  mdr,
+  rssi,
+  security: sec,
+  ssid,
+});
+
+class SoftAPService {
+  static configureWifi = async ({
+    channel = DEFAULT_WIFI_CHANNEL,
+    index = DEFAULT_WIFI_INDEX,
+    password,
+    security,
+    ssid,
+  }: WifiNetwork): Promise<void> => {
+    const publicKey = await SoftAPService._getPublicKey();
+    const encryptedPassword = password
+      ? publicKey.encrypt(password, 'hex')
+      : '';
+
+    const { r: responseCode } = await fetchJSON({
```

Comment:
Why don't you use `fetch` and look at the HTTP response code?

Reply:
since there's the code in the response, it can be more errors come from photon even if http response code is 200. Or it doesn't make sense to put this code in the json response from particle side.

Reply:
So the photon will always return a 200?

Reply:
I mean even if it will return 200, it still can be an error there.